app/src/main/java/io/homeassistant/companion/android/sensors/SensorReceiver.kt

```diff
@@ -57,41 +58,35 @@ class SensorReceiver : BroadcastReceiver() {
         Intent.ACTION_POWER_DISCONNECTED
     )
 
+    private val skippableActions = mapOf(
+        "android.app.action.NEXT_ALARM_CLOCK_CHANGED" to NextAlarmManager.nextAlarm.id,
+        "android.bluetooth.device.action.ACL_CONNECTED" to BluetoothSensorManager.bluetoothConnection.id,
+        "android.bluetooth.device.action.ACL_DISCONNECTED" to BluetoothSensorManager.bluetoothConnection.id,
+        BluetoothAdapter.ACTION_STATE_CHANGED to BluetoothSensorManager.bluetoothState.id,
+        TelephonyManager.ACTION_PHONE_STATE_CHANGED to PhoneStateSensorManager.phoneState.id
+    )
+
     override fun onReceive(context: Context, intent: Intent) {
 
         DaggerSensorComponent.builder()
             .appComponent((context.applicationContext as GraphComponentAccessor).appComponent)
             .build()
             .inject(this)
 
-        when (intent.action) {
-            "android.app.action.NEXT_ALARM_CLOCK_CHANGED" -> {
-                val sensorDao = AppDatabase.getInstance(context).sensorDao()
-                val sensor = sensorDao.get(NextAlarmManager.nextAlarm.id)
-                if (sensor?.enabled != true) {
-                    Log.d(TAG, "Alarm Sensor disabled, skipping sensors update")
-                    return
-                }
-            }
-            "android.bluetooth.device.action.ACL_CONNECTED",
-                "android.bluetooth.device.action.ACL_DISCONNECTED" -> {
-                val sensorDao = AppDatabase.getInstance(context).sensorDao()
-                val sensorBtConn = sensorDao.get(BluetoothSensorManager.bluetoothConnection.id)
-                if (sensorBtConn?.enabled != true) {
-                    Log.d(TAG, "Bluetooth Connection Sensor disabled, skipping sensors update")
-                    return
-                }
-            }
-            BluetoothAdapter.ACTION_STATE_CHANGED -> {
-                val sensorDao = AppDatabase.getInstance(context).sensorDao()
-                val sensorBtState = sensorDao.get(BluetoothSensorManager.bluetoothState.id)
-                if (sensorBtState?.enabled != true) {
-                    Log.d(TAG, "Bluetooth State Sensor disabled, skipping sensors update")
-                    return
-                }
+        if (skippableActions.containsKey(intent.action)) {
+            val sensor = skippableActions[intent.action]
+            if (isSensorEnabled(context, sensor!!) != true) {
+                Log.d(TAG, String.format
+                    ("Sensor %s corresponding to received event %s is disabled, skipping sensors update", sensor, intent.action))
+                return
             }
         }
 
+        if (intent.action.equals(TelephonyManager.ACTION_PHONE_STATE_CHANGED) &&
+            isSensorEnabled(context, PhoneStateSensorManager.callNumber.id)!!) {
+            PhoneStateSensorManager().updateCallNumber(context, intent)
```

Comment:
This does not trigger an actual update to get sent to HA, that happens in `updateSensors` down below

Reply:
Yep, update will trigger right after that